src/lib/notifications/notifications.ts

```diff
@@ -42,38 +41,52 @@ export async function requestPermissionsAndRegisterToken(
   }
 }
 
-export function registerTokenChangeHandler(
-  getAgent: () => BskyAgent,
-  account: SessionAccount,
-): () => void {
-  // listens for new changes to the push token
-  // In rare situations, a push token may be changed by the push notification service while the app is running. When a token is rolled, the old one becomes invalid and sending notifications to it will fail. A push token listener will let you handle this situation gracefully by registering the new token with your backend right away.
-  const sub = Notifications.addPushTokenListener(async newToken => {
-    logger.debug(
-      'Notifications: Push token changed',
-      {tokenType: newToken.data, token: newToken.type},
-      logger.DebugContext.notifications,
-    )
-    try {
-      await getAgent().api.app.bsky.notification.registerPush({
-        serviceDid: SERVICE_DID(account.service),
-        platform: devicePlatform,
-        token: newToken.data,
-        appId: 'xyz.blueskyweb.app',
-      })
-      logger.debug(
-        'Notifications: Sent push token (event)',
-        {
-          tokenType: newToken.type,
-          token: newToken.data,
-        },
-        logger.DebugContext.notifications,
-      )
-    } catch (error) {
-      logger.error('Notifications: Failed to set push token', {message: error})
+export function useNotificationsRegistration() {
+  const [currentPermissions] = Notifications.usePermissions()
+  const {getAgent} = useAgent()
+  const {currentAccount} = useSession()
+
+  const prevAccountDid = React.useRef<string | undefined>(undefined)
+
+  React.useEffect(() => {
+    if (
+      !currentPermissions?.granted ||
+      !currentAccount ||
+      prevAccountDid.current === currentAccount.did
```

Comment:
I am not actually certain why we are using `prevAccountDid` (or before, it was called `currentAccountDid`). I assume that maybe `getAgent` can change? I couldn't get this to happen when I removed this check but decided to leave it incase there was a reason.

Reply:
We should remove this check, it’s from the old days before we started remounting the tree. You can assume `currentAccount` never changes from the tree’s perspective.

Reply:
So the whole notion of prevAccountDid should be deleted.